db/db.go

```diff
@@ -596,6 +596,20 @@ func GetValidatorDeposits(publicKey []byte) (*types.ValidatorDeposits, error) {
 		deposits.LastEth1DepositTs = deposits.Eth1Deposits[len(deposits.Eth1Deposits)-1].BlockTs
 	}
 
+	// retrieve address names from bigtable
+	names := make(map[string]string)
+	for _, v := range deposits.Eth1Deposits {
+		names[string(v.FromAddress)] = ""
+	}
+	names, _, err = BigtableClient.GetAddressesNamesArMetadata(&names, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	for k, v := range deposits.Eth1Deposits {
+		deposits.Eth1Deposits[k].FromName = names[string(v.FromAddress)]
+	}
+
```

Comment:
**Suggestion:** All this code should be placed in the if above:
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/e92a323f8087281e9b9b908467db7026d85388b1/db/db.go#L595-L597

Yes, there SHOULD not be a validator without deposit but still it would increase readability of the code.